app/models/two_factor_authentication_removal.rb

```diff
@@ -0,0 +1,45 @@
+class TwoFactorAuthenticationRemoval < ApplicationRecord
+  belongs_to :publisher
+  # 2 Weeks represented in seconds
+  TWO_FACTOR_AUTHENTICATION_REMOVAL_WAITING_PERIOD = 1209600
+  # 4 Weeks represented in seconds
+  LOCKED_STATUS_WAITING_PERIOD = 2592000
+
+  # 6 Weeks represented in seconds
+  TOTAL_WAITING_PERIOD = TWO_FACTOR_AUTHENTICATION_REMOVAL_WAITING_PERIOD + LOCKED_STATUS_WAITING_PERIOD
+
+  # Returns time remaining in seconds
+  def total_time_remaining
+    time = Time.now
```

Comment:
Prefer to be more ruby-like to simply be like `TOTAL_WAITING_PERIOD - (Time.now - created_at)`